test/testMC.sh

```diff
@@ -46,159 +48,171 @@
+}
+
+function printErr {
+		echo "!!! ${THIS_SCRIPT}: ${BASH_LINENO[0]}: error: ${1}"
+		exit 1
+}
+
+function testStep {
+		printInfo "Test ${1} ..."
+		echo "    executing: ${2}"
+		if eval ${2}; then
+				printSucc "${1} was successful."
+				echo
+		else
+				printErr "${1} was not successful. Aborting."
+		fi
+}
+
+
 ### BEGIN PREPARATION ###
 
 if [ -d "${ROOTPWA}" ]; then
-	echo "Installation of RootPwa found in: ${ROOTPWA}"
-	echo "Starting MC test ..."
+	printInfo "Installation of ROOTPWA found in: ${ROOTPWA}"
+	echo "    Starting MC test ..."
 else
-	echo "No installation of RootPwa found. Aborting..."
-	exit 1
+	printErr "No installation of ROOTPWA found. Aborting."
 fi
 
-TESTDIR=${PWD}/ROOTPWA_MC_TEST
-
 if [ -d "${TESTDIR}" ]; then
-	echo "Test directory ${TESTDIR} already exists. Aborting..."
-	exit 1
+	printErr "Test directory ${TESTDIR} already exists. Aborting."
 fi
 
-echo "Creating directories ..."
+printInfo "Creating directories ..."
 
-mkdir "${TESTDIR}"
-mkdir "${TESTDIR}/data"
-mkdir "${TESTDIR}/weighted_mc_data"
-mkdir "${TESTDIR}/keyfiles"
-mkdir "${TESTDIR}/reference_fit"
-mkdir "${TESTDIR}/fits"
-mkdir "${TESTDIR}/log"
+mkdir --verbose "${TESTDIR}"
+mkdir --verbose "${TESTDIR}/data"
+mkdir --verbose "${TESTDIR}/weighted_mc_data"
+mkdir --verbose "${TESTDIR}/keyfiles"
+mkdir --verbose "${TESTDIR}/reference_fit"
+mkdir --verbose "${TESTDIR}/fits"
+mkdir --verbose "${TESTDIR}/log"
 
-echo "Test directory and subfolders created."
-echo "Changing to test directory."
+echo "    Test directory and subfolders created."
+printInfo "Changing to test directory ${TESTDIR}."
 
 cd "${TESTDIR}"
 
 exec > >(tee -i ./log/logfile.txt)
 exec 2>&1
 
-echo "Copying necessary files to test directory ..."
+printInfo "Copying necessary files to test directory ..."
 
-cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/generator_noBeamSimulation.conf" "./"
-cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/wavelist.compass.2008.88waves" "./"
-cp "${ROOTPWA}/rootpwa.config" "./"
-cp "${ROOTPWA}/test/mcTest/reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root.example" "./reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root"
+cp --verbose "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/generator_noBeamSimulation.conf" "./"
+cp --verbose "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/wavelist.compass.2008.88waves" "./"
+cp --verbose "${ROOTPWA}/rootpwa.config" "./"
+cp --verbose "${ROOTPWA}/test/mcTest/reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root.example" "./reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root"
 
 # make sure the integral binning in the config file is set correctly
-sed -i.bak 's/^integralBinning.*$/integralBinning                        = [ { "mass": (1.8, 1.82) } ]/' rootpwa.config
-rm -f rootpwa.config.bak
+sed -i 's/^integralBinning.*$/integralBinning                        = [ { "mass": (1.8, 1.82) } ]/' rootpwa.config
```

Comment:
Could you please write `sed -i '' 's/^integralBinning...`. `sed` (probably `getopt`) in BSD (and thus MacOS) does not support missing arguments to options.